presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationGroupedExecution.java

```diff
@@ -0,0 +1,181 @@
+                " partition_by_hash_buckets = 2" +
+                ")");
+
+        assertUpdate("INSERT INTO test_grouped_execution VALUES (0, 0, 0), (0, 1, 1), (1, 0, 1)", 3);
+        assertQuery(getSession(), "SELECT key1, COUNT(1) FROM test_grouped_execution GROUP BY key1", "SELECT * FROM (VALUES (0, 2), (1, 1))", assertRemoteExchangesCount(1));
+
+        assertUpdate("DROP TABLE test_grouped_execution");
+    }
+
+    @Test
+    public void testGroupedExecutionMultiLevelPartitioning()
+    {
+        assertUpdate("CREATE TABLE IF NOT EXISTS test_grouped_execution_mtlvl (" +
+                "key1 BIGINT WITH (primary_key=true)," +
+                "key2 BIGINT WITH (primary_key=true)," +
+                "key3 BIGINT WITH (primary_key=true)," +
+                "key4 BIGINT WITH (primary_key=true)," +
+                "attr1 BIGINT" +
+                ") WITH (" +
+                " partition_by_hash_columns = ARRAY['key1', 'key2']," +
+                " partition_by_hash_buckets = 2," +
+                " partition_by_second_hash_columns = ARRAY['key3']," +
+                " partition_by_second_hash_buckets = 2" +
+                ")");
+
+        assertUpdate("INSERT INTO test_grouped_execution_mtlvl VALUES (0, 0, 0, 0, 0), (0, 0, 0, 1, 1), (1, 1, 1, 0, 0), (1, 1, 1, 1, 1)", 4);
+        assertQuery(getSession(), "SELECT key1, key2, key3, COUNT(1) FROM test_grouped_execution_mtlvl GROUP BY key1, key2, key3", "SELECT * FROM (VALUES (0, 0, 0, 2), (1, 1, 1, 2))", assertRemoteExchangesCount(1));
+
+        assertUpdate("DROP TABLE test_grouped_execution_mtlvl");
+    }
+
+    @Test
+    public void testGroupedExecutionMultiLevelCombinedPartitioning()
+    {
+        assertUpdate("CREATE TABLE test_grouped_execution_hash_range (" +
+                "key1 BIGINT WITH (primary_key=true)," +
+                "key2 BIGINT WITH (primary_key=true)," +
+                "key3 BIGINT WITH (primary_key=true)," +
+                "key4 BIGINT WITH (primary_key=true)," +
+                "attr1 BIGINT" +
+                ") WITH (" +
+                "  partition_by_hash_columns = ARRAY['key1']," +
+                "  partition_by_hash_buckets = 2," +
+                "  partition_by_second_hash_columns = ARRAY['key2']," +
+                "  partition_by_second_hash_buckets = 3," +
+                "  partition_by_range_columns = ARRAY['key3']," +
+                "  range_partitions = '[{\"lower\": null, \"upper\": \"4\"}, {\"lower\": \"4\", \"upper\": \"9\"}, {\"lower\": \"9\", \"upper\": null}]'" +
+                ")");
+
+        assertUpdate("INSERT INTO test_grouped_execution_hash_range VALUES (0, 0, 0, 0, 0), (0, 0, 9, 0, 9), (0, 0, 9, 1, 0), (1, 1, 0, 0, 1), (1, 1, 9, 0, 2)", 5);
+        assertQuery(getSession(), "SELECT key1, key2, key3, COUNT(1) FROM test_grouped_execution_hash_range GROUP BY key1, key2, key3", "SELECT * FROM (VALUES (0, 0, 0, 1), (0, 0, 9, 2), (1, 1, 0, 1), (1, 1, 9, 1))", assertRemoteExchangesCount(1));
```

Comment:
```suggestion
        assertQuery(
            getSession(), 
            "SELECT key1, key2, key3, COUNT(1) FROM test_grouped_execution_hash_range GROUP BY key1, key2, key3", 
            "VALUES (0, 0, 0, 1), (0, 0, 9, 2), (1, 1, 0, 1), (1, 1, 9, 1)", 
            assertRemoteExchangesCount(1));
```

Apply same comment to all other tests

Reply:
Did it!